core/iwasm/common/wasm_native.h

```diff
@@ -68,6 +68,32 @@ bool
 wasm_native_unregister_natives(const char *module_name,
                                NativeSymbol *native_symbols);
 
+#if WASM_ENABLE_MODULE_INST_CONTEXT != 0
+void *
+wasm_native_create_context_key(void (*dtor)(wasm_module_inst_t inst,
```

Comment:
Had better use `WASMModuleInstanceCommon *` instead of `wasm_module_inst_t`

Reply:
what are the principles?

Reply:
Hi, in the original design, the wasm_module_t/wasm_module_inst_t/wasm_exec_env_t/.. are only used only in wasm_export.h, native libraries (libc-builtin, libc-wasi and more in core/iwasm/libraries) and main.c, and the detailed structure name (WASMModule/WASMModuleInstance/WASMExecEnv/..) are used in runtime internal. Though there may be some code inside runtime using the the formers, we had better obey the rules.

Reply:
done